neural_compressor/benchmark.py

```diff
@@ -135,391 +135,257 @@ def get_bounded_threads(core_ids, threads, sockets):
-            self.framework = get_model_fwk_name(user_model)
-            if self.framework == "tensorflow":
-                from .model.tensorflow_model import get_model_type
-                if get_model_type(user_model) == 'keras' and cfg.backend == 'itex':
-                    self.framework = 'keras'
-            if self.framework == "pytorch":
-                if cfg.backend == "default":
-                    self.framework = "pytorch_fx"
-                elif cfg.backend == "ipex":
-                    self.framework = "pytorch_ipex"
-                    import intel_extension_for_pytorch
-            cfg.framework = self.framework
-
-        if not isinstance(user_model, BaseModel):
-            logger.warning("Force convert framework model to neural_compressor model.")
-            if "tensorflow" in self.framework or self.framework == "keras":
-                self._model = NCModel(user_model, backend=self.framework, device=cfg.device)
-            else:
-                self._model = NCModel(user_model, backend=self.framework)
-        else:
-            # It is config of neural_compressor version < 2.0, no need in 2.0
-            if cfg.framework == "pytorch_ipex":
-                from neural_compressor.model.torch_model import IPEXModel
-                if not isinstance(user_model, IPEXModel):
-                    self._model = NCModel(user_model.model, framework=cfg.framework)
-                    return
-            self._model = user_model
-
-        if 'tensorflow' in self.framework:
-            self._model.name = cfg.model_name
-            self._model.output_tensor_names = cfg.outputs
-            self._model.input_tensor_names = cfg.inputs
-            self._model.workspace_path = options.workspace
-
-    def __repr__(self):
-        """Get the object representation in string format."""
-        return 'Benchmark'
-
-def fit(model, config=None, b_dataloader=None, b_func=None):
+    Args:
+        raw_cmd (string):           The commend to be benchmarked.
+        conf (BenchmarkConfig): The configuration for benchmark containing accuracy goal,
+                                  tuning objective and preferred calibration & quantization
+                                  tuning space etc.
+
+    Example::
+
+        # Run benchmark according to config
+        from neural_compressor.benchmark import fit_with_raw_cmd
+
+        conf = BenchmarkConfig(iteration=100, cores_per_instance=4, num_of_instance=7)
+        fit_with_raw_cmd("test.py", conf)
+    """
+    if conf is not None:
+        if conf.backend == "ipex":
+            import intel_extension_for_pytorch
+        assert sys.platform in ['linux', 'win32'], 'only support platform windows and linux...'
+        # disable multi-instance for running bechmark on GPU device
+        set_all_env_var(conf)
+
+    config_instance(raw_cmd)
+    summary_benchmark()
+
+
+def fit(model, conf, b_dataloader=None, b_func=None):
```

Comment:
is it incompatible change as previous parameter is config, but now it's conf.

and it's inconsistent between those user facing APIs, like Quantization.fit & MixedPrecision.fit. some of them are Optional, but some of them are not

Reply:
now, all fit functions use conf, because quantization used conf, so we align it to conf.